### DISCUSSION TITLE
Any reason why I can get both "No difference detected" and "drift detected"?

### DISCUSSION THREAD

**Discussion by katangafor** [prisma/prisma#20070] [Q&A] [answered]
I dropped a migration file somewhere in a giant merge conflict. Luckily, prisma migrate caught it, and gave me the ol' `drift detected` error message.

So I thought I'd generate some SQL to cover the difference using `prisma migrate diff --from-... --to-... --script`".

I put that in my migrations folder, with the right timestamp-ish folder name, and again ran `prisma migrate diff` to make sure it worked. I see `No difference detected.`. Hooray!

But now, while running `prisma migrate dev`, I'm still getting `drift detected`. I'm pretty confused how this is happening, since I figured `prisma migrate dev` and `prisma migrate diff` both just detected differences between the DB and the shadow DB (after applying existing migrations to the shadow DB). But when I inspect the shadow DB, it looks identical to my regular DB!

Am I missing a step after creating the migration file? Do I need to do something with the _prisma_migrations table?

**Comment by ludralph** [comment]
Hi @katangafor 👋 

Can you please take a look at this section of the documentation and follow the steps mentioned there?

- https://www.prisma.io/docs/guides/migrate/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute

If it doesn't fix it,  you could provide a minimal reproduction then we can run it locally and check what's going on.

**Reply by nurul3101** [reply, marked as answer]
@katangafor Did you had a chance to look at Raphael's response? Please let us know in case you still need assistance. If not, please consider marking the discussion as answered/closed.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
